From 7c510cd20bb443064ab4a0d2c6d917e807b2671b Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 22 May 2023 19:45:28 -0700
Subject: [PATCH] agent/meson: Replace filename with basename

This avoids emitting absolute paths into generated sourcecode, it
improves reproducibility. See [1]

[1] https://developer-old.gnome.org/gobject/stable/glib-mkenums.html

Upstream-Status: Submitted [https://gitlab.freedesktop.org/libnice/libnice/-/merge_requests/258]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 agent/meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/agent/meson.build b/agent/meson.build
index 2e5b272..44fcd6b 100644
--- a/agent/meson.build
+++ b/agent/meson.build
@@ -29,14 +29,14 @@ gnome = import('gnome')
 
 agent_enum_types_c =  gnome.mkenums('agent-enum-types.c', sources : agent_headers,
 				    fhead: '#include <config.h>\n#include <glib-object.h>\n#include "agent.h"\n#include "pseudotcp.h"\n#include "agent-enum-types.h"',
-				    fprod: '\n/* enumerations from "@filename@" */',
+				    fprod: '\n/* enumerations from "@basename@" */',
 				    vhead: 'GType\n@enum_name@_get_type (void)\n{\n  static GType type = 0;\n  if (!type) {\n    static const G@Type@Value values[] = {',
 				    vprod: '      { @VALUENAME@, "@VALUENAME@", "@valuenick@" },',
 				    vtail: '      { 0, NULL, NULL }\n    };\n    type = g_@type@_register_static ("@EnumName@", values);\n  }\n  return type;\n}\n\n')
 
 agent_enum_types_h = gnome.mkenums('agent-enum-types.h', sources : agent_headers,
 				   fhead: '#ifndef __AGENT_ENUM_TYPES_H__\n#define __AGENT_ENUM_TYPES_H__ 1\n\n#include <glib-object.h>\n\nG_BEGIN_DECLS\n',
-				   fprod: '/* enumerations from "@filename@" */\n',
+				   fprod: '/* enumerations from "@basename@" */\n',
 				   vhead: 'GType @enum_name@_get_type (void) G_GNUC_CONST;\n#define NICE_TYPE_@ENUMSHORT@ (@enum_name@_get_type())\n',
 					  ftail: 'G_END_DECLS\n\n#endif /* !AGENT_ENUM_TYPES_H */')
 
-- 
2.40.1

